Framework/Core/Windows/Devices/Spi/SpiDevice.cs

```diff
@@ -0,0 +1,285 @@
+        public void Read(byte[] buffer)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentException();
+            }
+
+            TransferInternal(null, buffer, false);
+        }
+
+        /// <summary>
+        /// Transfer data sequentially to the device.
+        /// </summary>
+        /// <param name="writeBuffer">Array containing data to write to the device.</param>
+        /// <param name="readBuffer">Array containing data read from the device.</param>
+        public void TransferSequential(byte[] writeBuffer, byte[] readBuffer)
+        {
+            if ((writeBuffer == null) || (readBuffer == null))
+            {
+                throw new ArgumentException();
+            }
+
+            TransferInternal(writeBuffer, readBuffer, false);
+        }
+
+        /// <summary>
+        /// Transfer data using a full duplex communication system. Full duplex allows both the master and the slave to
+        /// communicate simultaneously.
+        /// </summary>
+        /// <param name="writeBuffer">Array containing data to write to the device.</param>
+        /// <param name="readBuffer">Array containing data read from the device.</param>
+        public void TransferFullDuplex(byte[] writeBuffer, byte[] readBuffer)
+        {
+            if ((writeBuffer == null) || (readBuffer == null))
+            {
+                throw new ArgumentException();
+            }
+
+            TransferInternal(writeBuffer, readBuffer, true);
+        }
+
+        /// <summary>
+        /// Closes the connection to the device.
+        /// </summary>
+        public void Dispose()
+        {
+            lock (m_syncLock)
+            {
+                if (!m_disposed)
+                {
+                    Dispose(true);
+                    GC.SuppressFinalize(this);
+                    m_disposed = true;
+                }
+            }
+        }
+
+        internal static string[] GetValidBusNames()
+        {
+            return new string[] {
+                s_SpiPrefix + "1",
+                s_SpiPrefix + "2",
+                s_SpiPrefix + "3",
+                s_SpiPrefix + "4",
+            };
+        }
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        extern private void InitNative();
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        extern private void DisposeNative();
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        extern private void TransferInternal(byte[] writeBuffer, byte[] readBuffer, bool fullDuplex);
+
+        /// <summary>
+        /// Releases internal resources held by the device.
+        /// </summary>
+        /// <param name="disposing">True if called from Dispose, false if called from the finalizer.</param>
+        private void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                lock (m_syncLock)

```

Comment:
you can remove the lock
the only called of this one is already locking (Dispose()) and the other caller is on a system thread that knows better